kratos/utilities/variable_utils.h

```diff
@@ -800,24 +800,22 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
     {
         KRATOS_TRY
 
-        if(rNodes.size() != 0) {
-            // First we do a check
+        if (rNodes.size() != 0) {
+            // checking the first node to avoid error being thrown in parallel region
+            KRATOS_ERROR_IF_NOT(rNodes.begin()->HasDofFor(rVar)) << "Trying to fix/free dof of variable " << rVar.Name() << " but this dof does not exist!" << std::endl;
             CheckVariableExists(rVar, rNodes);
 
```

Comment:
@roigcarlo suggested to add a loop that check the existence of the DOF in all nodes in `FullDebug` mode.

Reply:
done